### ISSUE TITLE
results and discussion

### ISSUE THREAD

**Issue by menelaosas** [menelaosas/opsyedai_chatbot_Thesis_Demokritos#7]
- [ ] Experimental Results
- [ ] Inference Load Tests
- [ ] Speedup Attempts
- [ ] Performance Analysis
- [ ] Software Infrastructure
- [ ] Model Infrastructure
- [ ] Development Tooling

**Comment by menelaosas** [comment]
llama trainning  

<img width="312" height="227" alt="Image" src="https://github.com/user-attachments/assets/20a9a7bb-c29d-4b34-a51f-b05aa2040cd4" />

**Comment by menelaosas** [comment]
llama trainning
GPU Memory Status:
  Total:     8.18 GB
  Reserved:  7.85 GB
  Allocated: 5.74 GB
  Free:      0.34 GB
==================================================

==================================================
TRAINING CONFIGURATION
==================================================
Batch size: 1
Gradient accumulation: 8
Effective batch size: 8
Learning rate: 5e-06
Epochs: 5
Optimizer: OptimizerNames.PAGED_ADAMW_8BIT
Mixed precision (FP16): False
Gradient checkpointing: True
/home/menelaosas/Documents/Thesis_OPSYEDAI_chatbot/opsyed-ai-repo/notebooks/llm_trainning_original.py:239: FutureWarning: `tokenizer` is deprecated and will be removed in version 5.0.0 for `Trainer.__init__`. Use `processing_class` instead.
  trainer = Trainer(

GPU status before training:
==================================================
GPU Memory Status:
  Total:     8.18 GB
  Reserved:  7.85 GB
  Allocated: 5.74 GB
  Free:      0.34 GB
==================================================

==================================================
STARTING TRAINING
==================================================
This will take several hours on RTX 4070...
Monitor GPU memory usage with: nvidia-smi -l 1
The tokenizer has new PAD/BOS/EOS tokens that differ from the model config and generation config. The model config and generation config were aligned accordingly, being updated with the tokenizer's values. Updated tokens: {'eos_token_id': 128009, 'pad_token_id': 128009}.
{'loss': 2.6604, 'grad_norm': 2.9889140129089355, 'learning_rate': 3e-06, 'epoch': 0.17}                                                                                                                                                                         
{'loss': 2.6376, 'grad_norm': 2.5985424518585205, 'learning_rate': 4.997390310845578e-06, 'epoch': 0.35}                                                                                                                                                         
{'loss': 2.5669, 'grad_norm': 2.5136911869049072, 'learning_rate': 4.968093843200407e-06, 'epoch': 0.52}                                                                                                                                                         
{'loss': 2.4645, 'grad_norm': 2.4867770671844482, 'learning_rate': 4.9066219978460485e-06, 'epoch': 0.69}                                                                                                                                                        
{'loss': 2.3422, 'grad_norm': 2.426178216934204, 'learning_rate': 4.813776154295767e-06, 'epoch': 0.87}                                                                                                                                                          
{'loss': 2.2136, 'grad_norm': 2.4275543689727783, 'learning_rate': 4.690766700109659e-06, 'epoch': 1.03}                                                                                                                                                         
{'loss': 2.083, 'grad_norm': 2.490053653717041, 'learning_rate': 4.5391972516417545e-06, 'epoch': 1.21}                                                                                                                                                          
{'loss': 1.9515, 'grad_norm': 2.37371563911438, 'learning_rate': 4.36104374844843e-06, 'epoch': 1.38}                                                                                                                                                            
{'loss': 1.824, 'grad_norm': 2.143308401107788, 'learning_rate': 4.15862869389448e-06, 'epoch': 1.56}                                                                                                                                                            
{'loss': 1.7056, 'grad_norm': 1.9630697965621948, 'learning_rate': 3.934590877769944e-06, 'epoch': 1.73}                                                                                                                                                         
{'loss': 1.604, 'grad_norm': 1.9223113059997559, 'learning_rate': 3.6918509756296876e-06, 'epoch': 1.9}                                                                                                                                                          
{'loss': 1.5064, 'grad_norm': 1.9463483095169067, 'learning_rate': 3.4335734733209457e-06, 'epoch': 2.07}                                                                                                                                                        
{'loss': 1.4121, 'grad_norm': 2.047849416732788, 'learning_rate': 3.1631254130708446e-06, 'epoch': 2.24}                                                                                                                                                         
{'loss': 1.3155, 'grad_norm': 2.2708146572113037, 'learning_rate': 2.8840324989417488e-06, 'epoch': 2.42}                                                                                                                                                        
{'loss': 1.2187, 'grad_norm': 2.5407257080078125, 'learning_rate': 2.599933133886934e-06, 'epoch': 2.59}                                                                                                                                                         
{'loss': 1.119, 'grad_norm': 2.549544334411621, 'learning_rate': 2.31453098760387e-06, 'epoch': 2.76}                                                                                                                                                            
{'loss': 1.0286, 'grad_norm': 2.5189201831817627, 'learning_rate': 2.031546713535688e-06, 'epoch': 2.94}                                                                                                                                                         
{'loss': 0.9522, 'grad_norm': 2.5791966915130615, 'learning_rate': 1.7546694444635394e-06, 'epoch': 3.1}                                                                                                                                                         
{'loss': 0.8844, 'grad_norm': 2.6063387393951416, 'learning_rate': 1.487508699018987e-06, 'epoch': 3.28}                                                                                                                                                         
{'loss': 0.8256, 'grad_norm': 2.699697494506836, 'learning_rate': 1.2335473260886046e-06, 'epoch': 3.45}                                                                                                                                                         
{'loss': 0.7748, 'grad_norm': 2.6405904293060303, 'learning_rate': 9.960961005524033e-07, 'epoch': 3.62}                                                                                                                                                         
{'loss': 0.7337, 'grad_norm': 2.5779335498809814, 'learning_rate': 7.782505622700964e-07, 'epoch': 3.8}                                                                                                                                                          
{'loss': 0.7031, 'grad_norm': 2.537240743637085, 'learning_rate': 5.828506609850054e-07, 'epoch': 3.97}                                                                                                                                                          
{'loss': 0.6792, 'grad_norm': 2.506282091140747, 'learning_rate': 4.1244373323601874e-07, 'epoch': 4.14}                                                                                                                                                         
{'loss': 0.6593, 'grad_norm': 2.5094614028930664, 'learning_rate': 2.6925129393015196e-07, 'epoch': 4.31}                                                                                                                                                        
{'loss': 0.6501, 'grad_norm': 2.480170488357544, 'learning_rate': 1.5514007549836979e-07, 'epoch': 4.49}                                                                                                                                                         
{'loss': 0.6447, 'grad_norm': 2.503558397293091, 'learning_rate': 7.159769218354873e-08, 'epoch': 4.66}                                                                                                                                                          
{'loss': 0.6445, 'grad_norm': 2.7255842685699463, 'learning_rate': 1.9713246713805588e-08, 'epoch': 4.83}                                                                                                                                                        
{'loss': 0.6433, 'grad_norm': 2.527204751968384, 'learning_rate': 1.6313218287128396e-10, 'epoch': 5.0}                                                                                                                                                          
{'train_runtime': 1748.3165, 'train_samples_per_second': 1.318, 'train_steps_per_second': 0.166, 'train_loss': 1.3947866982427137, 'epoch': 5.0}                                                                                                                 
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 290/290 [29:08<00:00,  6.03s/it]

Saving model to ../saved_models/llama 2025-10-26 12:35:49

==================================================
✅ TRAINING COMPLETED SUCCESSFULLY!
==================================================

Final GPU status:
==================================================
GPU Memory Status:
  Total:     8.18 GB
  Reserved:  6.52 GB
  Allocated: 5.78 GB
